security/advancedtls/advancedtls.go

```diff
@@ -139,21 +178,14 @@ func (o *ServerOptions) config() (*tls.Config, error) {
 	if o.Certificates == nil && o.GetCertificate == nil {
 		return nil, fmt.Errorf("either Certificates or GetCertificate must be specified")
 	}
-	if o.RequireClientCert && o.GetRootCAs == nil && o.RootCACerts == nil {
-		return nil, fmt.Errorf("server needs to provide root CA certs if requiring client cert")
+	if o.RequireClientCert && o.GetRootCAs == nil && o.RootCACerts == nil && o.VerifyPeer == nil {
```

Comment:
My understanding is that if the user chooses `SkipVerification`, then no verification mechanisms need to be provided, right? If that's the case, this `if` condition doesn't take it into account.

Reply:
The logic to "invoke" `SkipVerification` is in function `buildVerifyFunc`. Here(`Config` function) we are just doing a sanity check. But yeah I noticed that we shouldn't always ask users to provide verification checks because they have the options to choose `SkipVerification`.
I've modified this if statement to check against the scenario "user chooses `SkipVerification`, but doesn't provide any other custom verification" .